// gatsby-plugin-gatsby-cloud/ipc.js
"use strict";

exports.__esModule = true;
exports.emitRoutes = emitRoutes;
exports.emitRedirects = emitRedirects;
exports.emitRewrites = emitRewrites;
exports.emitHeaders = emitHeaders;
exports.emitFileNodes = emitFileNodes;

function emitRoutes(routes) {
  if (!process.send) {
    return;
  }

  process.send({
    type: `LOG_ACTION`,
    action: {
      type: `CREATE_ROUTE`,
      payload: {
        routes
      }
    }
  });
}

function emitRedirects(redirect) {
  if (!process.send) {
    return;
  }

  process.send({
    type: `LOG_ACTION`,
    action: {
      type: `CREATE_REDIRECT_ENTRY`,
      payload: redirect
    }
  });
}

function emitRewrites(rewrite) {
  if (!process.send) {
    return;
  }

  process.send({
    type: `LOG_ACTION`,
    action: {
      type: `CREATE_REWRITE_ENTRY`,
      payload: rewrite
    }
  });
}

function emitHeaders(header) {
  if (!process.send) {
    return;
  }

  process.send({
    type: `LOG_ACTION`,
    action: {
      type: `CREATE_HEADER_ENTRY`,
      payload: header
    }
  });
}

function emitFileNodes(file) {
  if (!process.send) {
    return;
  }

  process.send({
    type: `LOG_ACTION`,
    action: {
      type: `CREATE_FILE_NODE`,
      payload: file
    }
  });
}
